src/rum/instrumentation/DdRumReactNavigationTracking.tsx

```diff
@@ -66,12 +66,24 @@ export default class DdRumReactNavigationTracking {
     private static resolveNavigationStateChangeListener(): NavigationListener {
         if (DdRumReactNavigationTracking.navigationStateChangeListener == null) {
             DdRumReactNavigationTracking.navigationStateChangeListener = (event: EventArg<string, boolean, any>) => {
-                let nestedRoute = event.data?.state?.routes[event.data?.state?.index];
-                while (nestedRoute.state != undefined) {
-                    nestedRoute = nestedRoute.state.routes[nestedRoute.state.index];
+                let route = event.data?.state?.routes[event.data?.state?.index];
+
+                if (route == undefined) {
+                    // RUMM-1400 in some cases the route seem to be undefined
+                    console.log("navigation state changed with undefined route");
+                    return
+                }
+
+                while (route.state != undefined) {
+                    let nestedRoute = route.state.routes[route.state.index];
+                    if (nestedRoute == undefined) {
+                        // RUMM-1400 in some cases the route seem to be undefined
```

Comment:
You are not logging this here but you do in the upper level. Maybe you could extract this check in a `isValidRoute` method and reuse it in both places.